lib/file-source/src/file_server.rs

```diff
@@ -23,6 +24,19 @@ use tokio::time::delay_for;
 
 use crate::paths_provider::PathsProvider;
 
+#[derive(Debug, Snafu)]
+pub enum FileServerError<
```

Comment:
If this enum isn't used outside of this source, it doesn't need to be `pub`.

Reply:
It is used in the return type of a public function (`run`) and therefore can't be kept private:

```
private type `FileServerError<<PP as PathsProvider>::Error, <C as futures::Sink<Vec<(bytes::Bytes, std::string::String)>>>::Error>` in public interface
can't leak private typerustcE0446
```